src/test/kotlin/automaton/constructor/model/algorithms/HellingsAlgoTests.kt

```diff
@@ -0,0 +1,374 @@
+        grammar.productions.add(Production(A94, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A120, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A116, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A84, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A80, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A226, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A227, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A21, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A3, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A111, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A219, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(S, mutableListOf()))
+        grammar.productions.add(Production(A287, mutableListOf(A80, A129)))
+        grammar.productions.add(Production(A287, mutableListOf(A84, A201)))
+        grammar.productions.add(Production(A287, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A279, mutableListOf(A80, A129)))
+        grammar.productions.add(Production(A279, mutableListOf(A84, A201)))
+        grammar.productions.add(Production(A279, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(A71, mutableListOf(A262, A165)))
+        grammar.productions.add(Production(A71, mutableListOf(A263, A183)))
+        grammar.productions.add(Production(A71, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(A63, mutableListOf(A262, A165)))
+        grammar.productions.add(Production(A63, mutableListOf(A263, A183)))
+        grammar.productions.add(Production(A63, mutableListOf(Terminal('1'))))
+        grammar.productions.add(Production(S, mutableListOf(A94, A63)))
+        grammar.productions.add(Production(S, mutableListOf(A106, A279)))
+        grammar.productions.add(Production(S, mutableListOf(A94, A71)))
+        grammar.productions.add(Production(S, mutableListOf(A106, A287)))
+        grammar.productions.add(Production(U, mutableListOf(Terminal('0'))))
+        grammar.productions.add(Production(U326, mutableListOf(Terminal('1'))))
+
+        val expectedWithSAsNonterminal = setOf(
+            "S, S0, S0",
+            "S, S2, S2",
+            "S, S3, S3",
+            "S, S5, S5",
+            "S, S6, S6",
+            "S, S4, S4",
+            "S, S1, S1",
+            "S, S4, S6",
+            "S, S2, S4",
+            "S, S0, S2",
+            "S, S1, S5",
+            "S, S0, S6"
+        )
+        val actualWithSAsNonterminal = execute(automaton, grammar).filter {
+            it.nonterminal == S
+        }.map { it.toString() }.toSet()
+        assertTrue(actualWithSAsNonterminal.equals(expectedWithSAsNonterminal))
```

Comment:
`assertEquals` позволяет в случае падения тестов получать сообщений информативнее, чем `expected: true, actual: false`. 
```suggestion
        assertEquals(expectedWithSAsNonterminal, actualWithSAsNonterminal)
```